Объединим данные о погоде и о поломках оборудования. Это необходимо сделать по столбцам "Дата открытия" и "data".

In [56]:
import pandas as pd

In [57]:
df_requests = pd.read_csv('requests.csv')
df_requests.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25074 entries, 0 to 25073
Data columns (total 13 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   Unnamed: 0             25074 non-null  int64 
 1   № заявки               25074 non-null  object
 2   Цех                    25074 non-null  object
 3   Инв. номер             25074 non-null  object
 4   Год выпуска            25074 non-null  int64 
 5   Оборудование           25074 non-null  object
 6   Модель                 25074 non-null  object
 7   Дата открытия          25074 non-null  object
 8   Описание неисправ.     25074 non-null  object
 9   Причина неисправности  15711 non-null  object
 10  Принятые меры          21851 non-null  object
 11  Дата закрытия заявки   25074 non-null  object
 12  длительность ремонта   25074 non-null  object
dtypes: int64(2), object(11)
memory usage: 2.5+ MB


In [58]:
df_weather = pd.read_csv('weather.csv')
df_weather.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2191 entries, 0 to 2190
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Unnamed: 0  2191 non-null   int64 
 1   day         2191 non-null   int64 
 2   month       2191 non-null   int64 
 3   year        2191 non-null   int64 
 4   temp        2191 non-null   int64 
 5   data        2191 non-null   object
dtypes: int64(5), object(1)
memory usage: 102.8+ KB


In [59]:
df_requests['Дата открытия'] = df_requests['Дата открытия'].astype('datetime64[ns]')
df_requests['Дата закрытия заявки'] = df_requests['Дата закрытия заявки'].astype('datetime64[ns]')
df_requests.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25074 entries, 0 to 25073
Data columns (total 13 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   Unnamed: 0             25074 non-null  int64         
 1   № заявки               25074 non-null  object        
 2   Цех                    25074 non-null  object        
 3   Инв. номер             25074 non-null  object        
 4   Год выпуска            25074 non-null  int64         
 5   Оборудование           25074 non-null  object        
 6   Модель                 25074 non-null  object        
 7   Дата открытия          25074 non-null  datetime64[ns]
 8   Описание неисправ.     25074 non-null  object        
 9   Причина неисправности  15711 non-null  object        
 10  Принятые меры          21851 non-null  object        
 11  Дата закрытия заявки   25074 non-null  datetime64[ns]
 12  длительность ремонта   25074 non-null  object        
dtypes

In [60]:
df_weather['data'] = df_weather['data'].astype('datetime64[ns]')
df_weather.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2191 entries, 0 to 2190
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype         
---  ------      --------------  -----         
 0   Unnamed: 0  2191 non-null   int64         
 1   day         2191 non-null   int64         
 2   month       2191 non-null   int64         
 3   year        2191 non-null   int64         
 4   temp        2191 non-null   int64         
 5   data        2191 non-null   datetime64[ns]
dtypes: datetime64[ns](1), int64(5)
memory usage: 102.8 KB


In [61]:
df_requests['data'] = df_requests['Дата открытия']
df_requests.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25074 entries, 0 to 25073
Data columns (total 14 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   Unnamed: 0             25074 non-null  int64         
 1   № заявки               25074 non-null  object        
 2   Цех                    25074 non-null  object        
 3   Инв. номер             25074 non-null  object        
 4   Год выпуска            25074 non-null  int64         
 5   Оборудование           25074 non-null  object        
 6   Модель                 25074 non-null  object        
 7   Дата открытия          25074 non-null  datetime64[ns]
 8   Описание неисправ.     25074 non-null  object        
 9   Причина неисправности  15711 non-null  object        
 10  Принятые меры          21851 non-null  object        
 11  Дата закрытия заявки   25074 non-null  datetime64[ns]
 12  длительность ремонта   25074 non-null  object        
 13  d

In [62]:
df_merged = df_requests.merge(df_weather, on='data', how='outer')
df_merged

Unnamed: 0_x   № заявки           Цех Инв. номер  Год выпуска  \
0           23977.0  18-04-176  ВМЗ Цех № 44    А-5-158       1987.0   
1           23147.0  12-11-112        ЦЕХ 43   Т-1/1864       1995.0   
2           22472.0  12-11-109        ЦЕХ 43    Ш-2-261       2002.0   
3           21111.0  12-11-118        ЦЕХ 26     А-4/47       1961.0   
4           24043.0  12-11-111        ЦЕХ 43    Ф-5/699       1991.0   
...             ...        ...           ...        ...          ...   
25813           NaN        NaN           NaN        NaN          NaN   
25814           NaN        NaN           NaN        NaN          NaN   
25815           NaN        NaN           NaN        NaN          NaN   
25816           NaN        NaN           NaN        NaN          NaN   
25817           NaN        NaN           NaN        NaN          NaN   

                          Оборудование     Модель Дата открытия  \
0          фpeзep.beptиk.cпeциaл.с чпу  MA655C630    2008-04-26   
1         токарно-винтор. станок с чпу     MDW-10    2011-12-21   
2      станок координатно-шлифовальный    S55-400    2012-01-11   
3         координатно-расточной станок        2А3    2012-02-07   
4                 фрезерhый ст-к с чпу   MH-1000S    2012-03-12   
...                                ...        ...           ...   
25813                              NaN        NaN           NaT   
25814                              NaN        NaN           NaT   
25815                              NaN        NaN           NaT   
25816                              NaN        NaN           NaT   
25817                              NaN        NaN           NaT   

            Описание неисправ. Причина неисправности          Принятые меры  \
0               повышенный шум                   NaN                    NaN   
1            закупка  з/частей                   NaN       закупка з/частей   
2              закупка матрицы                   NaN        закупка матрицы   
3         закупка оптич.атчика                   NaN  закупка оптич.датчика   
4      закупка платы изм.cист.                   NaN          закупка платы   
...                        ...                   ...                    ...   
25813                      NaN                   NaN                    NaN   
25814                      NaN                   NaN                    NaN   
25815                      NaN                   NaN                    NaN   
25816                      NaN                   NaN                    NaN   
25817                      NaN                   NaN                    NaN   

      Дата закрытия заявки длительность ремонта       data  Unnamed: 0_y  \
0               2024-12-13   6075 days 00:00:00 2008-04-26           NaN   
1               2012-11-15    330 days 00:00:00 2011-12-21           NaN   
2               2012-11-15    309 days 00:00:00 2012-01-11           NaN   
3               2012-11-15    282 days 00:00:00 2012-02-07           NaN   
4               2012-11-15    248 days 00:00:00 2012-03-12           NaN   
...                    ...                  ...        ...           ...   
25813                  NaT                  NaN 2024-12-27        2186.0   
25814                  NaT                  NaN 2024-12-28        2187.0   
25815                  NaT                  NaN 2024-12-29        2188.0   
25816                  NaT                  NaN 2024-12-30        2189.0   
25817                  NaT                  NaN 2024-12-31        2190.0   

        day  month    year  temp  
0       NaN    NaN     NaN   NaN  
1       NaN    NaN     NaN   NaN  
2       NaN    NaN     NaN   NaN  
3       NaN    NaN     NaN   NaN  
4       NaN    NaN     NaN   NaN  
...     ...    ...     ...   ...  
25813  27.0   12.0  2024.0   1.0  
25814  28.0   12.0  2024.0   1.0  
25815  29.0   12.0  2024.0   2.0  
25816  30.0   12.0  2024.0  -2.0  
25817  31.0   12.0  2024.0  -3.0  

[25818 rows x 19 columns]

In [63]:
df_merged = df_merged.drop(['Unnamed: 0_x', 'Unnamed: 0_y'], axis=1)

In [64]:
df_merged.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25818 entries, 0 to 25817
Data columns (total 17 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   № заявки               25074 non-null  object        
 1   Цех                    25074 non-null  object        
 2   Инв. номер             25074 non-null  object        
 3   Год выпуска            25074 non-null  float64       
 4   Оборудование           25074 non-null  object        
 5   Модель                 25074 non-null  object        
 6   Дата открытия          25074 non-null  datetime64[ns]
 7   Описание неисправ.     25074 non-null  object        
 8   Причина неисправности  15711 non-null  object        
 9   Принятые меры          21851 non-null  object        
 10  Дата закрытия заявки   25074 non-null  datetime64[ns]
 11  длительность ремонта   25074 non-null  object        
 12  data                   25818 non-null  datetime64[ns]
 13  d

Удалим данные, где отсутствуют температура.

In [65]:
df_merged = df_merged.dropna(subset=['temp'])
df_merged.info()

<class 'pandas.core.frame.DataFrame'>
Index: 11279 entries, 14532 to 25817
Data columns (total 17 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   № заявки               10535 non-null  object        
 1   Цех                    10535 non-null  object        
 2   Инв. номер             10535 non-null  object        
 3   Год выпуска            10535 non-null  float64       
 4   Оборудование           10535 non-null  object        
 5   Модель                 10535 non-null  object        
 6   Дата открытия          10535 non-null  datetime64[ns]
 7   Описание неисправ.     10535 non-null  object        
 8   Причина неисправности  10342 non-null  object        
 9   Принятые меры          10375 non-null  object        
 10  Дата закрытия заявки   10535 non-null  datetime64[ns]
 11  длительность ремонта   10535 non-null  object        
 12  data                   11279 non-null  datetime64[ns]
 13  da

In [66]:
df_merged = df_merged.dropna(subset=['Цех'])
df_merged.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10535 entries, 14540 to 25769
Data columns (total 17 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   № заявки               10535 non-null  object        
 1   Цех                    10535 non-null  object        
 2   Инв. номер             10535 non-null  object        
 3   Год выпуска            10535 non-null  float64       
 4   Оборудование           10535 non-null  object        
 5   Модель                 10535 non-null  object        
 6   Дата открытия          10535 non-null  datetime64[ns]
 7   Описание неисправ.     10535 non-null  object        
 8   Причина неисправности  10342 non-null  object        
 9   Принятые меры          10375 non-null  object        
 10  Дата закрытия заявки   10535 non-null  datetime64[ns]
 11  длительность ремонта   10535 non-null  object        
 12  data                   10535 non-null  datetime64[ns]
 13  da

После всех преобразований, у нас осталось 10535 строк с данными о выходе из строя оборудования с дневной температурой.

In [67]:
df_merged = df_merged.drop(['длительность ремонта'], axis=1)
df_merged.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10535 entries, 14540 to 25769
Data columns (total 16 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   № заявки               10535 non-null  object        
 1   Цех                    10535 non-null  object        
 2   Инв. номер             10535 non-null  object        
 3   Год выпуска            10535 non-null  float64       
 4   Оборудование           10535 non-null  object        
 5   Модель                 10535 non-null  object        
 6   Дата открытия          10535 non-null  datetime64[ns]
 7   Описание неисправ.     10535 non-null  object        
 8   Причина неисправности  10342 non-null  object        
 9   Принятые меры          10375 non-null  object        
 10  Дата закрытия заявки   10535 non-null  datetime64[ns]
 11  data                   10535 non-null  datetime64[ns]
 12  day                    10535 non-null  float64       
 13  mo

In [68]:
df_merged['длительность ремонта'] = df_merged['Дата закрытия заявки'] - df_merged['Дата открытия']
df_merged

№ заявки     Цех Инв. номер  Год выпуска  \
14540  19-01-006  ЦЕХ 22     Р-5/40       1989.0   
14541  19-01-011  ЦЕХ 23    Ф-5/675       1993.0   
14542  19-01-015  ЦЕХ 16    Э-9/822       1985.0   
14543  19-01-017  ЦЕХ 26    Т-5-778       2013.0   
14544  19-01-007  ЦЕХ 16    Г-3/633       2000.0   
...          ...     ...        ...          ...   
25765  24-11-051  ЦЕХ 23    А-5-221       2007.0   
25766  24-11-046  ЦЕХ 22   Т-1-2698       2007.0   
25767  24-11-047  ЦЕХ 15    Г-3/537       1985.0   
25768  24-11-048  ЦЕХ 15    Г-3/538       1985.0   
25769  24-11-054  ЦЕХ 20    У-6-622       2017.0   

                                 Оборудование              Модель  \
14540               tokapho-peboльb.ct. с чпу          1В340Ф30РМ   
14541      вертикально-фрезерный станок с чпу              6М13НК   
14542                        источник питания            ИСВУ-315   
14543                    обрабатывающий центр               E160A   
14544                       печь вертикальная           6818-2185   
...                                       ...                 ...   
25765  5-ти координатный обрабатывающий центр         go-Mill 350   
25766                     токарhый ст-к с чпу  SCHAUBLIN-140R CNC   
25767                   эл.пeчь kamep.bakуmh.            FHV90GHS   
25768                   эл.пeчь kamep.bakуmh.            FHV90GHS   
25769  установка хромирование газовым методом             УМДП-И2   

      Дата открытия                      Описание неисправ.  \
14540    2019-01-09                          не запускается   
14541    2019-01-09                       нет движения по z   
14542    2019-01-09                     замкнуло плату, дым   
14543    2019-01-09                          не выходит в 0   
14544    2019-01-09                                  не вкл   
...             ...                                     ...   
25765    2024-11-11  ош 04, 403, 419, шторка не закрывается   
25766    2024-11-11   не держит размер, шум при перемещении   
25767    2024-11-11                         ремонт вак с-мы   
25768    2024-11-11                  проверка герметичности   
25769    2024-11-12                  проверка герметичности   

                                 Причина неисправности  \
14540                        выход из строя гл привода   
14541                                     ош оператора   
14542       вышел из строя электролит в релейном блоке   
14543  села батарейка, слетели параметры привязки осей   
14544                            неисправна плата тпчт   
...                                                ...   
25765  плохой контакт в замке шторки смены инструмента   
25766               раскрутилась затяжная гайка на швп   
25767                                заварена трещина.   
25768                              проверка выполнена.   
25769                    замена вакуумного уплотнения.   

                                           Принятые меры Дата закрытия заявки  \
14540                                  замена гл привода           2019-01-11   
14541                                        диагностика           2019-01-09   
14542                                 замена электролита           2019-01-11   
14543  замена батарейки, волсстановление параметров п...           2019-01-15   
14544                                       ремонт платы           2019-01-15   
...                                                  ...                  ...   
25765                    диагностика и регулировка замка           2024-11-13   
25766                             подтянули гайку на швп           2024-11-13   
25767                                  заварена трещина.           2024-11-14   
25768                                проверка выполнена.           2024-11-13   
25769                      замена вакуумного уплотнения.           2024-11-13   

            data   day  month    year  temp длительность ремонта  
14540 2019-01-09   9.0    1.0  2019.0 -10.0               2 

In [69]:
df_merged[['Категория оборудования', 'Инв. номер']] = df_merged['Инв. номер'].str.split('-', n=1, expand=True)
df_merged

№ заявки     Цех Инв. номер  Год выпуска  \
14540  19-01-006  ЦЕХ 22       5/40       1989.0   
14541  19-01-011  ЦЕХ 23      5/675       1993.0   
14542  19-01-015  ЦЕХ 16      9/822       1985.0   
14543  19-01-017  ЦЕХ 26      5-778       2013.0   
14544  19-01-007  ЦЕХ 16      3/633       2000.0   
...          ...     ...        ...          ...   
25765  24-11-051  ЦЕХ 23      5-221       2007.0   
25766  24-11-046  ЦЕХ 22     1-2698       2007.0   
25767  24-11-047  ЦЕХ 15      3/537       1985.0   
25768  24-11-048  ЦЕХ 15      3/538       1985.0   
25769  24-11-054  ЦЕХ 20      6-622       2017.0   

                                 Оборудование              Модель  \
14540               tokapho-peboльb.ct. с чпу          1В340Ф30РМ   
14541      вертикально-фрезерный станок с чпу              6М13НК   
14542                        источник питания            ИСВУ-315   
14543                    обрабатывающий центр               E160A   
14544                       печь вертикальная           6818-2185   
...                                       ...                 ...   
25765  5-ти координатный обрабатывающий центр         go-Mill 350   
25766                     токарhый ст-к с чпу  SCHAUBLIN-140R CNC   
25767                   эл.пeчь kamep.bakуmh.            FHV90GHS   
25768                   эл.пeчь kamep.bakуmh.            FHV90GHS   
25769  установка хромирование газовым методом             УМДП-И2   

      Дата открытия                      Описание неисправ.  \
14540    2019-01-09                          не запускается   
14541    2019-01-09                       нет движения по z   
14542    2019-01-09                     замкнуло плату, дым   
14543    2019-01-09                          не выходит в 0   
14544    2019-01-09                                  не вкл   
...             ...                                     ...   
25765    2024-11-11  ош 04, 403, 419, шторка не закрывается   
25766    2024-11-11   не держит размер, шум при перемещении   
25767    2024-11-11                         ремонт вак с-мы   
25768    2024-11-11                  проверка герметичности   
25769    2024-11-12                  проверка герметичности   

                                 Причина неисправности  \
14540                        выход из строя гл привода   
14541                                     ош оператора   
14542       вышел из строя электролит в релейном блоке   
14543  села батарейка, слетели параметры привязки осей   
14544                            неисправна плата тпчт   
...                                                ...   
25765  плохой контакт в замке шторки смены инструмента   
25766               раскрутилась затяжная гайка на швп   
25767                                заварена трещина.   
25768                              проверка выполнена.   
25769                    замена вакуумного уплотнения.   

                                           Принятые меры Дата закрытия заявки  \
14540                                  замена гл привода           2019-01-11   
14541                                        диагностика           2019-01-09   
14542                                 замена электролита           2019-01-11   
14543  замена батарейки, волсстановление параметров п...           2019-01-15   
14544                                       ремонт платы           2019-01-15   
...                                                  ...                  ...   
25765                    диагностика и регулировка замка           2024-11-13   
25766                             подтянули гайку на швп           2024-11-13   
25767                                  заварена трещина.           2024-11-14   
25768                                проверка выполнена.           2024-11-13   
25769                      замена вакуумного уплотнения.           2024-11-13   

            data   day  month    year  temp длительность ремонта  \
14540 2019-01-09   9.0    1.0  2019.0 -10.0               2

In [70]:
df_merged.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10535 entries, 14540 to 25769
Data columns (total 18 columns):
 #   Column                  Non-Null Count  Dtype          
---  ------                  --------------  -----          
 0   № заявки                10535 non-null  object         
 1   Цех                     10535 non-null  object         
 2   Инв. номер              10535 non-null  object         
 3   Год выпуска             10535 non-null  float64        
 4   Оборудование            10535 non-null  object         
 5   Модель                  10535 non-null  object         
 6   Дата открытия           10535 non-null  datetime64[ns] 
 7   Описание неисправ.      10535 non-null  object         
 8   Причина неисправности   10342 non-null  object         
 9   Принятые меры           10375 non-null  object         
 10  Дата закрытия заявки    10535 non-null  datetime64[ns] 
 11  data                    10535 non-null  datetime64[ns] 
 12  day                     10535 non

In [71]:
df_merged.to_csv('join.csv')